app/views/play/level/PlayLevelView.coffee

```diff
@@ -197,17 +197,42 @@ module.exports = class PlayLevelView extends RootView
       })
     @setupGod() if @waitingToSetUpGod
 
-    if features.china and e.level.get('slug') is 'magic-rush'
-      @checkForTournamentEnd()
+    if features.china
+      @levelSlug = e.level.get('slug')
+      @setupMandateCheck()
 
-  checkForTournamentEnd: =>
+  setupMandateCheck: =>
     return if @destroyed
     $.get '/db/mandate', (data) =>
       return if @destroyed
-      if data?[0]?.currentTournament isnt 'magic-rush'
-        window.location.href = '/play/ladder/magic-rush'
-      else
-        setTimeout @checkForTournamentEnd, 60 * 1000
+      courseInstanceID = @courseInstanceID or utils.getQueryVariable 'league'
+      if @isTournamentBlocked data?[0], courseInstanceID, @levelSlug
+        unless me.isAdmin()
+          window.location.href = '/play/ladder/'+@levelSlug+(if courseInstanceID then '/course/'+courseInstanceID else "")
+      setTimeout @setupMandateCheck, 60 * 1000
```

Comment:
Does everyone in the tournament need to request the server every minute in order to check the mandate. Isn't the mandate fairly static, and once there is a date offset it should be possible to check the date on the client side more frequently and maybe do the mandate check less frequently.

If this is simpler then we can do this.

Reply:
You are right. We don't have `startAt` and `endAt` before, so we need to check mandate every minute, to confirm if the tournament is end.

As we have start and end field now, I think we could request it less frequently while setup a new function to check it using local date only